### PR TITLE
Add conditional compilation of default firewall driver

### PR DESCRIPTION
**Pull request by mheon**
The NETAVARK_DEFAULT_FW environment variable controls the default firewall driver that will be used by the compiled Netavark. Currently supported values are "iptables" (the default, if the environment variable is unset), "nftables", and "none" (we'll add "firewalld" as a supported value once that driver is done).

Unsupported values result in a panic/failure to build.

### CODE REVIEW

**Comment by Luap99** [comment]
Also AFAIK the centos stream 10 spec is part of our repo so we need to edit the spec to set this for RHEL 10
@lsm5 Can you show us where to set `NETAVARK_DEFAULT_FW=nftables` env for the build in the spec?

File: src/commands/version.rs
Comment (mheon) [reply src/commands/version.rs:25]: Is this an either-or thing with your subsequent comment? Set it as an environment variable, or use the `#[cfg]` syntax?

Comment (Luap99) [reply src/commands/version.rs:25]: I would do both, set the the rustc-cfg and rusc-env var in build.rs to make the actual code simpler

**Review by Luap99 — APPROVED** [review]
LGTM

I leave it to @lsm5 and @jnovy to tell us if the upstream spec must be changed to make this changes for Centos stream 10/ RHEL10 but no reason to block this one here as it can happen in a follow up.

**Pull request merged**
